pandas/_libs/lib.pyx

```diff
@@ -600,6 +600,8 @@ def array_equivalent_object(ndarray left, ndarray right) -> bool:
                     if not array_equivalent(x, y):
                         return False
 
+            elif PyArray_Check(x) or PyArray_Check(y):
+                return False
```

Comment:
Below we would essentially do `bool(left == right)` which raises an error for arrays/lists of size > 1 (and that error would then be caught below, turned into `return False`. So for now, just directly returning False here when left or right is a numpy array and the other not.

That then avoids doing this `bool(left == right)` for empty array/list case, which triggers the warning.

Reply:
The first `if` just before is `if PyArray_Check(x) and PyArray_Check(y)`, so the `or` is sufficient